data_override/data_override.F90

```diff
@@ -361,13 +365,21 @@ subroutine read_table(data_table)
     type(data_type)  :: data_entry
 
     logical               :: ongrid
+    logical               :: table_exists !< Flag indicating existence of data_table
     character(len=128)    :: region, region_type
 
     integer :: sunit
 
 !  Read coupler_table
+    inquire(file='data_table', EXIST=table_exists)
+    if (.not. table_exists) then
+      call mpp_error(NOTE, 'data_override_mod: File data_table does not exist.')
+      table_size = 0
+      return
+    end if
+
     open(newunit=iunit, file='data_table', action='READ', iostat=io_status)
-    if(io_status/=0) call mpp_error(FATAL, 'data_override_mod: Error in opening file data_table')
+    if(io_status/=0) call mpp_error(FATAL, 'data_override_mod: Error in opening file data_table.')
```

Comment:
Should this be in the 
```fortran 
if (.not. table_exists) then
```
block?

Reply:
this is for errors that may arise when the table doesn't exist but there's a problem reading the file. Hopefully, the return in the if block would end the subroutine if the file is missing and it wouldn't even get to this point (unless return only works for functions?)